### PR TITLE
Enable students to join classrooms

### PR DESCRIPTION
**Pull request by simoneduca**
Add join action.
Add config routes.
Add JoinClassroom and JoinButton components.

### CODE REVIEW

**Comment by simoneduca** [comment]
This is ready to go. @eatyourgreens or @shaunanoordin would be great if you could have a look.
Especially regarding the effects of changing `NODE_ENV` var in the webpack config.

File: src/constants/config.json [hunk 1]
```diff
@@ -5,10 +5,14 @@
     "dataLayerIndex": 1
   },
   "eduAPI": {
-    "authToken": "64aa60e488a555e2ef8b7f9d1621ec7e430dcb8b8baa0d00b8460eca5660d188",
+    "testing-only-authToken": "64aa60e488a555e2ef8b7f9d1621ec7e430dcb8b8baa0d00b8460eca5660d188",
```
Comment (shaunanoordin) [inline src/constants/config.json:8]: I just noticed that eduAPI's `authToken` variable isn't referenced anywhere else in the app - is this for future use or am I missing something? (e.g. implicit in one of our Panoptes libs.)

Comment (simoneduca) [inline src/constants/config.json:8]: You're right, we can get rid of that.

File: src/constants/config.json [hunk 2]
```diff
@@ -5,10 +5,14 @@
...
     "root": "https://education.staging.zooniverse.org/",
     "students": "students/classrooms/",
     "teachers": "teachers/classrooms/"
   },
+  "routes": {
```
Comment (shaunanoordin) [inline src/constants/config.json:13]: Query: is there a reason we have similar `root` and `students` variable/values under `config.eduAPI` and `config.routes`?

**Comment by shaunanoordin** [comment]
LGTM! I only have minor comments in the config file, but they're non-blocking. I'm happy to merge.
- Code passes sanity checks with only minor comments about extra variables in config.json
- Functionality confirmed - students can access the Join URL.
- New student landing page confirmed.

PR Status: **Validated**
Action: **Merging**

File: src/constants/config.json (hunk 2 above)
Comment (simoneduca) [inline src/constants/config.json:13]: It's a coincidence that the routes are similar. But I wanted to make clear that `routes` is for internal, app use, while `eduAPI` is used for external, API use.

**Pull request merged**